lib/Dialect/AIEVec/Transforms/ConvertVectorToAIEVec.cpp

```diff
@@ -301,12 +301,12 @@ struct FoldBroadcastToFMAOp : public OpConversionPattern<aievec::FMAOp> {
     if (!extOp)
       return failure();
 
-    auto newLhs = extOp.getVector();
+    auto lhs = extOp.getVector();
     // XXX: We assume a 1D vector
     auto pos = extOp.getPosition();
     int64_t zstart = cast<IntegerAttr>(pos[0]).getInt();
     rewriter.replaceOpWithNewOp<aievec::FMAOp>(
-        fmaOp, fmaOp.getResult().getType(), newLhs, rhs, adaptor.getAcc(),
+        fmaOp, fmaOp.getResult().getType(), lhs, rhs, adaptor.getAcc(),
         /*xstart =*/"0", /*xoffsets =*/"0x76543210", adaptor.getXoffsetsHi(),
```

Comment:
We would better to generate offsets and compute start by a function due to different data type, because i8 and i16 types would have different xoffsets from i32 type. It seems we need to get xstart, xoffset, xoffset_hi, xstep, xsquare, zstart, zoffset, zoffset_hi, zstep, zsquare when generate FMAOp. The order here looks not correct to me.